packages/ui/src/components/primitives/message/Embed.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { MessageProps } from './props';
+import { cn } from 'packages/ui/src/utils/utils';
+import { BlueLink } from '../ui/blue-link';
+import { parse } from './markdown/render';
+
+const EmbedText = (props: {
+	children: React.ReactNode;
+	className?: string;
+}) => (
+	<span className={cn(`font-light text-primary`, props.className)}>
+		{props.children}
+	</span>
+);
+
+const EmbedField = async (
+	props: NonNullable<EmbedProps['embed']['fields']>[number],
+) => {
+	const nameHTML = await parse(props.name);
+	const valueHTML = await parse(props.value);
+
+	return (
+		<div className="flex flex-col gap-1">
+			<EmbedText className="font-bold">{nameHTML}</EmbedText>
+			<EmbedText>{valueHTML}</EmbedText>
+		</div>
+	);
+};
+
+export interface EmbedProps {
+	embed: NonNullable<MessageProps['message']['embeds']>[number];
+}
+
+const numberToHex = (number: number | undefined) => {
+	if (!number) return undefined;
+	const hex = number.toString(16);
+	return hex.length === 1 ? '0' + hex : hex;
+};
+
+export const Embed = async (props: EmbedProps) => {
+	const { embed } = props;
+	const embedAuthorNameHTML = await parse(embed.author?.name ?? '');
```

Comment:
Seems like the parsing should happen in the component

Reply:
Fixed!